debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlUnsignedIntegerConverter.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mysql;
+
+import java.math.BigDecimal;
+
+/**
+ * A converter API for MySQL Unsigned Integer types. It intends to convert any integer type value from binlong into the correct representation of unsigned numeric
+ * MySQL binlog stores unsigned numeric into this format: (insertion value - Maximum data type boundary - 1), therefore to calculate the correct unsigned numeric representation
+ * we will inverse the original calculation by applying this calculation: (insertion value + Maximum data type boundary + 1). Please see DBZ-228 for more info
+ *
+ * @author Omar Al-Safi
+ */
+public class MySqlUnsignedIntegerConverter {
+    /**
+     * Maximum values for Unsigned Integer Types. Needed in order to calculate actual value of an Unsigned Integer Types from binlog value.
+     * Reference to {@link https://dev.mysql.com/doc/refman/5.7/en/integer-types.html}
+     */
+    private static final Short TINYINT_MAX_VALUE = 255;
+    private static final Integer SMALLINT_MAX_VALUE = 65535;
+    private static final Integer MEDIUMINT_MAX_VALUE = 16777215;
+    private static final Long INT_MAX_VALUE = 4294967295L;
+    private static final BigDecimal BIGINT_MAX_VALUE = new BigDecimal("18446744073709551615");
+
+    /**
+     * Private constructor
+     */
+    private MySqlUnsignedIntegerConverter(){}
+
+    /**
+     * Convert original value insertion of type 'TINYINT' into the correct TINYINT UNSIGNED representation
+     * Note: Unsigned TINYINT (8-bit) is represented in 'Short' 16-bit data type. Reference: https://kafka.apache.org/0102/javadoc/org/apache/kafka/connect/data/Schema.Type.html
+     *
+     * @param originalNumber {@link Short} the original insertion value
+     * @return {@link Short} the correct representation of the original insertion value
+     */
+    public static Short convertUnsignedTinyint(Short originalNumber){
+        if (originalNumber < 0){
+            return (short) (originalNumber + TINYINT_MAX_VALUE + 1);
+        } else {
+            return originalNumber;
+        }
+    }
+
+    /**
+     * Convert original value insertion of type 'SMALLINT' into the correct SMALLINT UNSIGNED representation
+     * Note: Unsigned SMALLINT (16-bit) is represented in 'Integer' 32-bit data type. Reference: https://kafka.apache.org/0102/javadoc/org/apache/kafka/connect/data/Schema.Type.html
+     *
+     * @param originalNumber {@link Integer} the original insertion value
+     * @return {@link Integer} the correct representation of the original insertion value
+     */
+    public static Integer convertUnsignedSmallint(Integer originalNumber){
```

Comment:
Yes, I'm confident it works, but it could work a bit better actually :) I've just rebased and merged this PR, adding one more commit to avoid some of the auto-boxing. All tests pass and it should be alright, but if you could take another look at this last commit on master, that'd be great.